### PR TITLE
Redirect to app.electricitymap.org

### PR DESCRIPTION
**Pull request by corradio**
Manual testing was performed:

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:electricitymap.tmrow.co 'user-agent:facebookexternalhit'
HTTP/1.1 200 OK

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:www.electricitymap.org
HTTP/1.1 301 Moved Permanently
Location: https://app.electricitymap.org/?lang=de&solar=true

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:app.electricitymap.org
HTTP/1.1 200 OK

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:electricitymap.org
HTTP/1.1 301 Moved Permanently
Location: https://app.electricitymap.org/?lang=de&solar=true

### CODE REVIEW

File: web/server.js [hunk 1]
```diff
@@ -131,17 +131,15 @@ app.use(express.static(STATIC_PATH, { etag: true, maxAge: isProduction ? '24h' :
...
+  const isNonAppDomain = req.get('host') !== 'app.electricitymap.org';
   const isStaging = req.get('host').includes('staging');
+  const isFacebookRobot = (req.headers['user-agent'] || '').indexOf('facebookexternalhit') !== -1;
 
   // Redirect all non-facebook, non-staging, non-(www.* or *.tmrow.co)
-  if (!isStaging && (isNonWWW || isTmrowCo) && (req.headers['user-agent'] || '').indexOf('facebookexternalhit') == -1) {
-    res.redirect(301, `https://www.electricitymap.org${req.originalUrl}`);
+  // redirect everyone except the Facebook crawler,
+  // else, we will lose all likes
+  if (!isStaging && isProduction && isNonAppDomain && !isFacebookRobot) {
```
Comment (corradio) [inline web/server.js:141]: A bit of cleanup and simpler variables

File: web/server.js [hunk 2]
```diff
@@ -131,17 +131,15 @@ app.use(express.static(STATIC_PATH, { etag: true, maxAge: isProduction ? '24h' :
 
 // App routes (managed by React Router)
 app.use('/', (req, res) => {
-  // On electricitymap.tmrow.co,
-  // redirect everyone except the Facebook crawler,
-  // else, we will lose all likes
-  const isTmrowCo = req.get('host').indexOf('electricitymap.tmrow') !== -1;
-  const isNonWWW = req.get('host') === 'electricitymap.org'
-    || req.get('host') === 'live.electricitymap.org';
+  const isNonAppDomain = req.get('host') !== 'app.electricitymap.org';
```
Comment (corradio) [inline web/server.js:134]: Added a simpler variable instead of the `isNonWWW` which was confusing.

File: web/views/pages/index.ejs [hunk 3]
```diff
@@ -9,10 +9,11 @@
         <meta property="og:type" content="website" />
         <!-- It's import to keep the old og:url, else, likes will not accumulate properly -->
         <meta property="og:url" content="http://electricitymap.tmrow.co" />
+
         <meta property="og:locale" content="<%= FBLocale %>" />
-        <% for(var i=0; i<supportedFBLocales.length; i++) {%>
+<% for(var i=0; i<supportedFBLocales.length; i++) { -%>
```
Comment (corradio) [inline web/views/pages/index.ejs:14]: Without the `-` at the end, the loop adds newlines which makes the HTML cumbersome to inspect.
I also changed the indentation to produce proper well-indented HTML

File: web/views/pages/index.ejs [hunk 4]
```diff
@@ -59,14 +60,14 @@
...
-        <% } %>
-        <% if (typeof canonicalUrl != 'undefined') { %>
+<% } %>
+        <!-- Canonical -->
+<% if (typeof canonicalUrl != 'undefined') { -%>
         <link rel="alternate" href="<%- canonicalUrl.replace(/[?|&]lang=[^&]*/, '') %>" hreflang="x-default" />
         <link rel="canonical" href="<%- canonicalUrl %>" />
-        <% } %>
-
+<% } %>
```
Comment (corradio) [inline web/views/pages/index.ejs:70]: same here, no changes in behavior however.

**Review by skovhus — APPROVED** [review]

**Comment by corradio** [comment]
Manual testing was performed:

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:electricitymap.tmrow.co 'user-agent:facebookexternalhit'
HTTP/1.1 200 OK

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:www.electricitymap.org
HTTP/1.1 301 Moved Permanently
Location: https://app.electricitymap.org/?lang=de&solar=true

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:app.electricitymap.org
HTTP/1.1 200 OK

❯ http -h http://localhost:8000/\?lang\=de\&solar\=true Host:electricitymap.org
HTTP/1.1 301 Moved Permanently
Location: https://app.electricitymap.org/?lang=de&solar=true

**Comment by corradio** [comment]
Edit: Manual testing was performed (see https://github.com/tmrowco/electricitymap-contrib/pull/3399#issue-738553726)

**Pull request merged**